### ISSUE TITLE
Issue with ImageRecordReader(...) constructor

### ISSUE THREAD

**Issue by titubeta** [deeplearning4j/deeplearning4j#831]
Hi,
It seems that ImageRecordReader constructor is by default taking a -1 value for nChannels, therefore it reads gibberish values when it reads an image. 
In this code, https://github.com/deeplearning4j/deeplearning4j/blob/master/deeplearning4j-core/src/main/java/org/deeplearning4j/datasets/canova/RecordReaderDataSetIterator.java#L152
when the variable "record" is created, it contains some gibberish values instead of RGB value in the range 0-255

If you set nChannels=1 in the constructor explicitly i.e. ImageRecordReader(numRows, numColumns, 1, true, labels); program runs fine, but the values in the "record" variable still contains gibberish values. 

If you set nChannels=3 in the constructor explicitly i.e. mageRecordReader(numRows, numColumns, 1, true, labels); it throws an exception : java.lang.ArrayIndexOutOfBoundsException: 157, probably because here https://github.com/deeplearning4j/nd4j/blob/master/nd4j-api/src/main/java/org/nd4j/linalg/util/FeatureUtil.java#L38, 'index' is getting some large values 

Basically, it is not reading images properly and this issue should be resolved with high priority because if someone wants to test DL models on their datasets, it will be very hard to get any meaningful results.

Thanks!
Shehroz

**Comment by agibsonccc** [comment]
Can you migrate this to https://github.com/deeplearning4j/Canova/issues? Thanks!

**Comment by titubeta** [comment]
OK done!
I also did some more debugging and it appears the way labels are computed are causing problem. I added more info in my issue on canova.
